pkgs/development/tools/misc/binutils/default.nix

```diff
@@ -106,7 +104,15 @@ stdenv.mkDerivation {
       else ./mips64-default-n64.patch)
   ;
 
-  outputs = [ "out" "info" "man" ];
+  outputs = [ "out" "info" "man" "dev" ]
+  # Ideally we would like to always install 'lib' into a separate
+  # target. Unfortunately cross-compiled binutils installs libraries
+  # across both `$lib/lib/` and `$out/$target/lib` with a reference
+  # from $out to $lib. Probably a binutils bug: all libraries should go
+  # to $lib as binutils does not build target libraries. Let's make our
+  # life slightly simpler by installing everything into $out for
+  # cross-binutils.
+  ++ lib.optionals (targetPlatform == hostPlatform) [ "lib" ];
```

Comment:
In case anybody finds this useful.  This does actually work and brings back the `lib` output on cross.  But after looking back at it this amount of after-the-fact fixup probably ought to get pushed back into upstream `binutils` instead of going in nixpkgs.

```suggestion
  outputs = [ "out" "info" "man" "dev" "lib" ];
  # cross-compiled binutils installs itself like this, in order to
  # allow everything to be combined in /usr on a FHS distribution:
  #
  #   $out/
  #        bin                    = binaries, named $target-$toolname
  #        $target/
  #                bin            = symlinks: $toolname->$out/bin/$target-$toolname
  #        lib/
  #            ldscripts          = hostPlatform-independent ldscripts
  #        $host/
  #              $target/
  #                      lib/     = libbfd and libopcodes
  #                      include/ = targetPlatform-specific headers
  #   $lib/
  #        lib/                   = all other libraries
  #
  # Since nixpkgs splits at /nix/store/xxx..xx-$target-binutils/ and
  # creates a separate store directory for each `hostPlatform`,
  # don't need separate $target subdirectories below that level.
  postInstall = ''
    mv -v $out/${targetPlatform.config}/lib/ldscripts $lib/lib/
    rmdir $out/${targetPlatform.config}/lib

  '' + lib.optionalString (targetPlatform != hostPlatform) ''
    mv -v $out/${hostPlatform.config}/${targetPlatform.config}/include/* $dev/include/
    rmdir $out/${hostPlatform.config}/${targetPlatform.config}/include
    ln -Ts $dev $dev/${hostPlatform.config}

    mv -v $out/${hostPlatform.config}/${targetPlatform.config}/lib/* $lib/lib/
    rmdir $out/${hostPlatform.config}/${targetPlatform.config}/lib
    rmdir $out/${hostPlatform.config}/${targetPlatform.config}
    rmdir $out/${hostPlatform.config}
    ln -Ts . $out/${hostPlatform.config}

    # .la files embed the path to $out
    mkdir $out/lib
    mv $lib/lib/*.la $out/lib

  '' + ''
    # to prevent false $dev<->$out dependencies
    rm -f $dev/nix-support/propagaed-build-inputs

    # add $lib to the rpath; $out will be removed by `--shrink-rpath` in the fixupPhase
    find $lib/ -type f -exec patchelf --add-rpath $lib/ {} \;
  '';

```